prow/config/secrets_agent.go

```diff
@@ -23,17 +23,32 @@ import (
 	"time"
 
 	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
 )
 
+var _ oauth2.TokenSource = &SecretAgent{}
+
 // SecretAgent watches a path and automatically loads the secrets stored.
 type SecretAgent struct {
 	sync.Mutex
-	secretsMap map[string][]byte
+	secretsMap               map[string][]byte
+	secretPathForTokenSource string
 }
 
-// Start will begin polling the secret file at the path. If the first load
-// fails, Start with return the error and abort. Future load failures will log
-// the failure message but continue attempting to load.
+// NewSecretAgent creates a SecretAgent. This can be used if there is no need to monitor
+// files to update the secret value. Useful for testing.
+func NewSecretAgent(tokenKeys []string) *SecretAgent {
```

Comment:
This isn't used for testing. It's used when you want a static token source. Don't use this struct for a static token source, use a static token source.